### PR TITLE
BUG: Prevent segfault in roll_weighted_var

### LINKED ISSUE #46760: BUG: Series rolling .var with window>14, center=True, and any win_type crashes python with no error message
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

"""
Crash occurs often when:
    window >= 15, 
    center=True,
    win_type='triang' (or any win_type AFAIK),
    usually in the first few trials.

Crash does not result in an error message, python just quits.

Crash does not occur when:
    window <= 14
    win_type is not specified
    center=False
    .var() is changed to .mean()
"""

x = pd.Series(0)

N = 10

for i in range(N):
    print('Starting trial %s out of %s...' % (i+1,N), end='')
    result = x.rolling(window=16, center=True, win_type='triang').var()
    print('success')
```

### Issue Description

This script, if executed, causes python to simply stop (as if calling `quit()`) ~3/4ths of the time, usually when `i` > 1. 

Crash seems to occur often when:
- window >= 15, 
- center=True,
- win_type='triang' (or any win_type AFAIK),

Crash does not seem to occur when:
- window <= 14
- win_type is not specified
- center=False
- .var() is changed to .mean()

Example of repeatedly running the script:
```
>>python module1.py
Starting trial 1 out of 10...success
Starting trial 2 out of 10...success
Starting trial 3 out of 10...success
Starting trial 4 out of 10...success

>>python module1.py
Starting trial 1 out of 10...success
Starting trial 2 out of 10...success
Starting trial 3 out of 10...success

>>python module1.py
Starting trial 1 out of 10...success
Starting trial 2 out of 10...success
Starting trial 3 out of 10...success
Starting trial 4 out of 10...success
Starting trial 5 out of 10...success
Starting trial 6 out of 10...success
Starting trial 7 out of 10...success
Starting trial 8 out of 10...success
Starting trial 9 out of 10...success
Starting trial 10 out of 10...success
```
### Expected Behavior

The script should run reliably, or at least deterministically.

An error should appear if there is an issue, but instead python appears to just `quit()`/crash without any message.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 4bfe3d07b4858144c219b9346329027024102ab6
python           : 3.10.4.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19044
machine          : AMD64
processor        : AMD64 Family 25 Model 33 Stepping 0, AuthenticAMD
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252

pandas           : 1.4.2
numpy            : 1.22.3
pytz             : 2022.1
dateutil         : 2.8.2
pip              : 22.0.4
setuptools       : 58.1.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
markupsafe       : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.0
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by joseortiz3**
- closes #46760
- edit: I added a test to `test_win_type.py`
- I added an entry in the latest `doc/source/whatsnew/v1.5.0.rst` file because this fixes a bug

### CODE REVIEW

File: pandas/_libs/window/aggregations.pyx [hunk 1]
```diff
@@ -1592,7 +1592,7 @@ def roll_weighted_var(const float64_t[:] values, const float64_t[:] weights,
 
     with nogil:
 
-        for i in range(win_n):
+        for i in range(min(win_n, n)):
```
Comment (mroeschke) [inline pandas/_libs/window/aggregations.pyx:1595]: Will need a unit test for this change (in `test_win_type.py`). You can use the example in your bug report (without the iteration)

File: doc/source/whatsnew/v1.5.0.rst [hunk 2]
```diff
@@ -599,7 +599,7 @@ Groupby/resample/rolling
 - Bug in :meth:`GroupBy.cummax` with ``int64`` dtype with leading value being the smallest possible int64 (:issue:`46382`)
 - Bug in :meth:`GroupBy.max` with empty groups and ``uint64`` dtype incorrectly raising ``RuntimeError`` (:issue:`46408`)
 - Bug in :meth:`.GroupBy.apply` would fail when ``func`` was a string and args or kwargs were supplied (:issue:`46479`)
--
+- Bug in :meth:`.Rolling.var` would segfault when calculating weighted variance with large window size (:issue:`46760`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.5.0.rst:602]: Maybe mention specifically when the window size was larger than the data size.

**Comment by joseortiz3** [comment]
Ok, thanks.

I'm unsure how to verify that my test I added to `test_win_type.py` is working. I'm unfamiliar with implementing tests in projects like pandas, forgive me while I'm learning :) and thanks again for your help.

File: pandas/tests/window/test_win_type.py [hunk 3]
```diff
@@ -676,3 +676,13 @@ def test_cmov_window_special_linear_range(win_types_special, step):
         .mean(**kwds[win_types_special])
     )
     tm.assert_series_equal(xp, rs)
+
+
+@td.skip_if_no_scipy
+def test_weighted_var_big_window_no_segfault():
```
Comment (mroeschke) [inline pandas/tests/window/test_win_type.py:682]: Could you add the `win_types` fixture in the signature and assign `win_type=win_type`? Same with `center`

**Review by mroeschke — COMMENTED** [review]
One comment, otherwise looks good!

File: pandas/tests/window/test_win_type.py (hunk 3 above)
Comment (joseortiz3) [reply pandas/tests/window/test_win_type.py:682]: > Same with `center`

I'm confused. There is no fixture for `centers` yet in `test_win_type.py`. Do you want me to add one? like:

```python
@pytest.fixture(params=[True, False])
def centers(request):
    return request.param
    
@td.skip_if_no_scipy
def test_weighted_var_big_window_no_segfault(win_types, centers):
    # Github Issue #46772
    x = Series(0)
    result = x.rolling(window=16, center=centers, win_type=win_types).var()
    expected = Series(np.NaN)

    tm.assert_series_equal(result, expected)
```

Comment (mroeschke) [reply pandas/tests/window/test_win_type.py:682]: This is defined in `confest.py` in the directory above, so pytest will grab that fixture if you add `center` in the test function signature.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Awesome, thanks @joseortiz3 (the failure was unrelated)

**Pull request merged**